docs/guides/templates.md

```diff
@@ -0,0 +1,41 @@
+# Templates
+
+When a Function is created, an example implementation and a Function metadata file are written into the new Function's working directory.  Together, these files are referred to as the Function's Template.  Included are the templates 'http' and 'events' for each supported language runtime.
+
+These embedded templates are minimal by design.  The Function contains a minimum of external dependencies, and the 'func.yaml' defines a final environment within which the Funciton will execute that is devoid of any extraneous packages or services.
+
+To make use of more complex inital Function implementions, or to define runtime environments with arbitrarily complex requirements, the templates system is fully pluggable.
+
+## External Git Repositories
+
+When creating a new Function, a Git repository can be specified as the source for the template files.  For example, the Boson Project maintains a set of example Functions at https://github.com/boson-project/templates which can be used during project creation.
+
+For example, the Boson Project Templates repository contains an example "Hello World" Function implementation in each of the officially supported languages.  To use this template via the CLI, use the flags:
```

Comment:
Can we just document a hypothetical instead? I'd prefer not to have an example that appears to be broken because we don't have the repo in place.

Reply:
The repo is no indeed in place, so it will work.  I also expect to be using it as soon/  However, a hypothetical may still be better than a template which, when used, says "coming soon"?

Reply:
OK as I mentioned above - with the repo in place, all good